receiver/mysqlreceiver/client.go

```diff
@@ -152,6 +217,229 @@ func (c *mySQLClient) getIndexIoWaitsStats() ([]IndexIoWaitsStats, error) {
 	return stats, nil
 }
 
+func (c *mySQLClient) getReplicaStatusStats() ([]ReplicaStatusStats, error) {
+	query := "SHOW REPLICA STATUS"
+
+	rows, err := c.client.Query(query)
+	if err != nil {
+		me, ok := err.(*mysql.MySQLError)
+		if !ok {
+			return nil, err
+		}
+
+		if me.Number != 1064 {
+			return nil, err
+		}
+
+		// fallback to slave status
+		query = "SHOW SLAVE STATUS"
+		rows, err = c.client.Query(query)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	defer rows.Close()
+	cols, err := rows.Columns()
+	if err != nil {
+		return nil, err
+	}
+
+	var stats []ReplicaStatusStats
+	for rows.Next() {
+		var s ReplicaStatusStats
+		dest := []interface{}{}
+		for _, col := range cols {
+			switch strings.ToLower(col) {
+			case "replica_io_state":
+				fallthrough
+			case "slave_io_state":
+				dest = append(dest, &s.replicaIOState)
+			case "source_host":
+				fallthrough
```

Comment:
This is to support as many different implementation as possible. I navigated three different implementations so far:

- bitnami's mysql `show slave status`
- mysql documentation `show slave status`
- mysql documentation `show replica status`